### ISSUE TITLE
Crash: Failed to grow buffer

### ISSUE THREAD

**Issue by teymur291994** [SDWebImage/SDWebImage#3175]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name    |       ios
 Platform Version   |     14
 SDWebImage Version  |  5.8.3 
 Integration Method  | carthage
 Xcode Version           | Xcode 12.4
 Repro rate              | only once
 Repro with our demo prj | 
 Demo project link       | 

### Issue Description and Steps

Fatal Exception: NSMallocException
Failed to grow buffer
0  CoreFoundation                 0x18b18086c __exceptionPreprocess
1  libobjc.A.dylib                0x1a019bc50 objc_exception_throw
2  CoreFoundation                 0x18b1f6518 -[NSInvocation retainArguments].cold.1
3  CoreFoundation                 0x18b1ed77c __CFReallocationFailed
4  CoreFoundation                 0x18b095088 __CFSafelyReallocate
5  Foundation                     0x18c45d94c _NSMutableDataGrowBytes
6  Foundation                     0x18c3b7c70 -[NSConcreteMutableData appendBytes:length:]
7  Foundation                     0x18c5e8534 __49-[_NSDispatchData enumerateByteRangesUsingBlock:]_block_invoke
8  libdispatch.dylib              0x18ada39f8 _dispatch_data_apply
9  Foundation                     0x18c3fe698 -[_NSDispatchData enumerateByteRangesUsingBlock:]
10 Foundation                     0x18c4034f0 -[NSConcreteMutableData appendData:]
11 SDWebImage                     0x101c8a6fc -[SDWebImageDownloaderOperation URLSession:dataTask:didReceiveData:] + 364 (SDWebImageDownloaderOperation.m:364)
12 SDWebImage                     0x101c9e9f0 -[SDWebImageDownloader URLSession:dataTask:didReceiveData:] + 447 (SDWebImageDownloader.m:447)
13 CFNetwork                      0x18b959e1c _CFNetworkHTTPConnectionCacheSetLimit
14 libdispatch.dylib              0x18ad7324c _dispatch_call_block_and_release
15 libdispatch.dylib              0x18ad74db0 _dispatch_client_callout
16 libdispatch.dylib              0x18ad7c10c _dispatch_lane_serial_drain
17 libdispatch.dylib              0x18ad7cc90 _dispatch_lane_invoke
18 libdispatch.dylib              0x18ad86d78 _dispatch_workloop_worker_thread
19 libsystem_pthread.dylib        0x1d6cad814 _pthread_wqthread
20 libsystem_pthread.dylib        0x1d6cb476c start_wqthread

**Comment by NSKevin** [comment]
Anyone ????

**Comment by manjuy124** [comment]
any update on this??

**Comment by manjuy124** [comment]
Followed these configurations!
https://github.com/SDWebImage/SDWebImage/wiki/Common-Problems#optimization-for-large-images

Applying following configuration worked for us.

`SDImageCache.sharedImageCache.maxMemoryCost = 1024 * 1024 * 4 * 20;`

**Comment by ngochai84** [comment]
I've just got this issue again, 
i configured SDImageCache.shared.config.maxMemoryCost = 1024 * 1024 * 4 * 20  and using SDWebimage version 5.1
